toolchain/driver/driver.cpp

```diff
@@ -435,6 +440,11 @@ class Driver::CompilationUnit {
 
   // Check the parse tree and produce SemIR. Returns true on success.
   auto RunCheck(const SemIR::File& builtins) -> bool {
+    // Can be called when the file fails to load, so ensure there's source.
+    if (!source_) {
+      return false;
+    }
+
```

Comment:
I don't feel strongly about this, but it might be more consistent and general for this to check for `parse_tree_` (and for `RunLower` to check for `sem_ir_`, and for `RunCodeGen` to check for `module_`).

Reply:
The reason I went this path is because it's valid to call these if there's no source_, but invalid to call RunLower if there's no sem_ir (if anything, that should CARBON_CHECK)

Reply:
Added some CHECKs